### PR TITLE
Add support for u-blox EVK-ANNA-B4 which uses the nRF52833.

### PR DESCRIPTION
**Pull request by perssonmagnus**
This board is similar to the nrf52833dk_nrf52833 with
different pin assignments.

Tested with blinky, button, and Bluetooth peripheral_hr.

Signed-off-by: Magnus Persson <magnus.persson@u-blox.com>

### CODE REVIEW

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:179]:
```suggestion
   .. code-block:: devicetree

      &uart1 {
        compatible = "nordic,nrf-uarte";
        current-speed = <115200>;
        status = "okay";
        tx-pin = <14>;
        rx-pin = <16>;
      };
```

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:183]:
```suggestion
2. Get a reference to the UART1 instance using ``DEVICE_DT_GET(DT_NODELABEL(uart1))``
```

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:194]:
```suggestion
==================

To select the pin numbers for tx-pin and rx-pin::

   tx-pin = <pin_no>
```

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:205]: I think this information is redundant, specially now that we'll soon move to pinctrl where port/pin notation is used

File: boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig:17]: this is a non-minimal setting, so should likely be removed (applications can enable it as needed)

File: tests/drivers/adc/adc_api/src/test_adc.c
Comment (gmarull) [inline tests/drivers/adc/adc_api/src/test_adc.c:64]: please add this change in a separate commit

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:120]: repeated information here

**Review by gmarull — CHANGES_REQUESTED** [review]
Thanks for this addition, I have a few comments

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:179]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:183]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:194]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig:17]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: tests/drivers/adc/adc_api/src/test_adc.c
Comment (perssonmagnus) [reply tests/drivers/adc/adc_api/src/test_adc.c:64]: Previously we have added a number of boards (the ones above EVKANNAB4_NRF52833) where we made this change together with adding the board folder, but I guess it is nicer to have a separate commit. Will change this.

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:120]: Thanks, will change.

**Comment by perssonmagnus** [comment]
Thanks for your comments. See individual answers.

**Review by tejlmand — DISMISSED** [review]

File: tests/drivers/adc/adc_api/src/test_adc.c
Comment (perssonmagnus) [reply tests/drivers/adc/adc_api/src/test_adc.c:64]: Done.

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done.

**Comment by perssonmagnus** [comment]
Second patch set updates according to comments given for each individual comment plus removed a broken link in index.rst.

**Review by tejlmand — DISMISSED** [review]
lgtm

**Comment by mbolivar-nordic** [comment]
@perssonmagnus you will need to rebase to get green CI results; the documentation failure is not due to your PR but the fix was merged to the main branch since you last pushed.

@tejlmand please re-approve after that is done

**Comment by perssonmagnus** [comment]
@tejlmand @mbolivar-nordic Thanks! Rebased and also updated commit message slightly to confirm to contribuition guidelines.

**Review by tejlmand — DISMISSED** [review]
lgtm - reapproved

**Comment by mbolivar-nordic** [comment]
@gmarull this is green now; please review

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:174]: just fix this one, other will eventually get updated.

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (gmarull) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:178]: same, others will eventually get updated (we're now in the process of removing many unnecessary `device_get_binding()` usages)

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

**Review by tejlmand — APPROVED** [review]
lgtm - reapproved

**Comment by mbolivar-nordic** [comment]
@perssonmagnus please be aware that every time you push, your reviewers will need to re-approve.

**Comment by perssonmagnus** [comment]
> @perssonmagnus please be aware that every time you push, your reviewers will need to re-approve.

@mbolivar-nordic  yes I noticed. Sorry about that, just trying to follow https://docs.zephyrproject.org/latest/contribute/#contribution-workflow. 

Is it preferred to use the "Commit suggestion" buttons in the GUI for fixing further review comments once you have one approval? Will this remove the need for re-approval?

**Review by gmarull — CHANGES_REQUESTED** [review]
All in-tree boards have now been ported to pinctrl, so please do the same here. You can refer to https://github.com/zephyrproject-rtos/zephyr/pull/43926 for more details.

**Pull request closed without merging**
